pkg/component/component.go

```diff
@@ -371,104 +391,144 @@ func GetComponentSource(client *occlient.Client, componentName string, applicati
 	namespacedOpenShiftObject, err := util.NamespaceOpenShiftObject(componentName, applicationName)
 	if err != nil {
 		return errors.Wrapf(err, "unable to create namespaced name")
 	}
 
-	projectName := client.GetCurrentProjectName()
+	// Create annotations
+	annotations := map[string]string{componentSourceURLAnnotation: newSource}
+	annotations[componentSourceTypeAnnotation] = newSourceType
 
-	oldSourceType, _, err := GetComponentSource(client, componentName, applicationName, projectName)
+	// Component Type
+	componentImageType, err := GetComponentType(client, componentName, applicationName, projectName)
 	if err != nil {
-		return errors.Wrapf(err, "unable to get source of %s component", componentName)
+		return errors.Wrap(err, "unable to get component image type for updating")
 	}
 
+	// Parse componentImageType before adding to labels
+	_, imageName, imageTag, _, err := occlient.ParseImageName(componentImageType)
+	if err != nil {
+		return errors.Wrap(err, "unable to parse image name")
+	}
+
+	// Retrieve labels
+	// Save component type as label
+	labels := componentlabels.GetLabels(componentName, applicationName, true)
+	labels[componentlabels.ComponentTypeLabel] = imageName
+	labels[componentlabels.ComponentTypeVersion] = imageTag
+
+	// ObjectMetadata are the same for all generated objects
+	// Create common metadata that will be updated throughout all objects.
+	commonObjectMeta := metav1.ObjectMeta{
+		Name:        namespacedOpenShiftObject,
+		Labels:      labels,
+		Annotations: annotations,
+	}
+
+	// STEP 2. Determine what the new source is going to be
+
 	glog.V(4).Infof("Updating component %s, from %s to %s (%s).", componentName, oldSourceType, newSource, newSourceType)
 
 	if (oldSourceType == "local" || oldSourceType == "binary") && newSourceType == "git" {
 		// Steps to update component from local or binary to git
-		// - update odo annotations and labels to reflect changes
-		// - update BuildConfig source to have git repository https://example.com/myrepo as a source.
-		// - update and remove supervisor from dc
-		// - trigger build (wait for it to finish)
-
-		annotations := map[string]string{componentSourceURLAnnotation: newSource}
-		annotations[componentSourceTypeAnnotation] = newSourceType
-		err = client.UpdateBuildConfig(namespacedOpenShiftObject, projectName, newSource, annotations)
+		// 1. Create a BuildConfig
+		// 2. Update DeploymentConfig with the new image
+		// 3. Clean up
+		// 4. Build the application
+
+		// CreateBuildConfig here!
+		glog.V(4).Infof("Creating BuildConfig %s using imageName: %s for updating", namespacedOpenShiftObject, imageName)
+		err = client.CreateBuildConfig(commonObjectMeta, imageName, newSource)
 		if err != nil {
 			return errors.Wrapf(err, "unable to update BuildConfig  for %s component", componentName)
 		}
+
+		// Get the current BuildConfig since we're going to update DeploymentConfig
+		bc, err := client.GetBuildConfigFromName(namespacedOpenShiftObject, projectName)
```

Comment:
We can retrieve the build Config here from the above create call as suggested in https://github.com/redhat-developer/odo/pull/622/files#r222987807

Reply:
Ahhhh I see what you mean. I'm going to update it.